xshop/users/api/views.py

```diff
@@ -36,3 +37,10 @@ def post(self, request):
         resp_serializer = TokenResponseSerializer(data=resp_data)
         resp_serializer.is_valid()
         return Response(resp_serializer.data)
+
+
+class Logout(APIView):
+    def get(self, request, format=None):
+        if request.user.is_authenticated:
+            request.user.auth_token.delete()
+        return Response(status=status.HTTP_200_OK)
```

Comment:
- What if the user is not authenticated? should return 403, forbidden... (LoginRequiredMixin) https://docs.djangoproject.com/en/3.1/topics/auth/default/#the-loginrequired-mixin

- Also, what about the user's session? refer here to the Django auth's logout https://github.com/django/django/blob/9ee693bd6cf4074f04ec51c6f3cfe87cad392f12/django/contrib/auth/__init__.py#L138 call it after token removal...
```Python
from django.contrib.auth import logout as auth_logout

class Logout(APIView):
    def get(self, request, format=None):
        if request.user.is_authenticated:
            request.user.auth_token.delete()
            auth_logout(request.user)
        return Response(status=status.HTTP_200_OK)
```

Add tests to make sure:
- only authenticated users can access this view
- auth_token is actually deleted

Reply:
I added the requirements except the auth_token delete test, can not test it after logout but its deleted in the admin.